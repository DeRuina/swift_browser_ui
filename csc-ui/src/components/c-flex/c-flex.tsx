import { Component, Host, h } from '@stencil/core';
/**
 * Component which fills the remaining space within a flex container
 *
 * @group Layout
 * @slot - Content
 */
@Component({
  tag: 'c-flex',
  styleUrl: 'c-flex.scss',
  shadow: true,
})
export class CFlex {
  render() {
    return (
      <Host>
        <slot></slot>
      </Host>
    );
  }
}
