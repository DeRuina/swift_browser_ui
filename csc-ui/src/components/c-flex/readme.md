# c-flex

<!-- Auto Generated Below -->


## Overview

Component which fills the remaining space within a flex container

## Slots

| Slot | Description |
| ---- | ----------- |
|      | Content     |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
