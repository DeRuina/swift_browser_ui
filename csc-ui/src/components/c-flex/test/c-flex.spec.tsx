import { newSpecPage } from '@stencil/core/testing';
import { CFlex } from '../c-flex';

describe('c-flex', () => {
  it('renders default html', async () => {
    const page = await newSpecPage({
      components: [CFlex],
      html: `<c-flex><div><p>Hello world!</p></div></c-flex>`,
    });

    expect(page.root).toMatchSnapshot();
  });
});
