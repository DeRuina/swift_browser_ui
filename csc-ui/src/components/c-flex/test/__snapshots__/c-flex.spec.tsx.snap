// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-flex renders default html 1`] = `
<c-flex>
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <div>
    <p>
      Hello world!
    </p>
  </div>
</c-flex>
`;
